frontend/js/ci-index.js

```diff
@@ -1,78 +1,95 @@
 (async () => {
     try {
-        var api_data = await makeAPICall('/v1/ci/projects')
+        var api_data = await makeAPICall('/v1/ci/projects');
     } catch (err) {
-            showNotification('Could not get data from API', err);
-            return;
+        showNotification('Could not get data from API', err);
+        return;
     }
-    api_data.data.forEach(el => {
-
-        const repo = el[0]
-        const branch = el[1]
-        const workflow = el[2]
-        const source = el[3]
-        const last_run = el[4]
 
-        let uri_display = repo;
-        let uri = repo;
+    const projectsTableBody = document.querySelector('#projects-table tbody');
+    let currentRepoRow = null; // Track the current repository row
 
-        if (source == 'github'){
-            uri_display = `<i class="icon github"></i>${repo}`;
-            uri = `https://www.github.com/${repo}`;
-        } else if (source == 'gitlab'){
-            uri_display = `<i class="icon gitlab"></i>${repo}`;
-            uri = `https://www.gitlab.com/${repo}`;
-        } else if (source == 'bitbucket'){
-            uri_display = `<i class="icon bitbucket"></i>${repo}`;
-            uri = `https://bitbucket.com/${repo}`;
-        } 
-
-        uri_link = `${uri_display} <a href="${uri}"><i class="icon external alternate"></i></a>`;
+    api_data.data.forEach(el => {
+        const repo = el[0];
+        const branch = el[1];
+        const workflow_id = el[2];
+        const source = el[3];
+        const last_run = el[4];
+        let workflow_name = el[5];
 
+        if (workflow_name == '' || workflow_name == null) {
+            workflow_name = workflow_id;
+        }
 
-        // insert new accordion row if repository not known
-        let td_node = document.querySelector(`td[data-uri='${uri}_${branch}']`)
-        if (td_node == null || td_node == undefined) {
-            let row = document.querySelector('#projects-table tbody').insertRow()
-            row.innerHTML = `
-                <td data-uri="${uri}_${branch}">
+        // Check if it's a new repository
+        if (currentRepoRow === null || currentRepoRow.repo !== repo) {
+            // Create a row for the repository with an accordion
+            currentRepoRow = projectsTableBody.insertRow();
+            currentRepoRow.repo = repo;
+            currentRepoRow.innerHTML = `
+                <td>
                     <div class="ui accordion" style="width: 100%;">
-                      <div class="title">
-                        <i class="dropdown icon"></i> ${uri_link}
-                      </div>
-                      <div class="content">
-                      </div>
+                        <div class="title">
+                            <i class="dropdown icon"></i>
+                            ${getRepoLink(repo, source)}
```

Comment:
you are often reusing values in modfied format where I see no escaping before. What happens if source is already corrupted? will it not just display here?

Reply:
I was out of the loop as to where (and more importantly, why) values needed sanitizing - I thought it was only for strings in user-set variables, and mostly for display reasons. I didn't know of it in terms of a security risk for any maliciously set variable (including ones that users can set via direct API calls, instead of the automated ones by eco-ci)

Looking at it through that lens, I did see quite a few places where we weren't sanitizing values in ci.js and ci-index.js, and I've updated those now.